presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java

```diff
@@ -535,13 +686,47 @@ else if (isNotNull) {
             if (presentStream != null && nullsAllowed) {
                 nulls = ensureCapacity(nulls, positionCount);
             }
+            dataLength = totalLength;
             data = ensureCapacity(data, totalLength);
             offsets = ensureCapacity(offsets, totalPositions + 1);
         }
         else {
+            dataLength = maxLength;
```

Comment:
Isn't this case has maxLength as 0 ?

Reply:
@arunthirupathi `outputRequired == false` doesn't mean the maxLength is 0. `outputRequired` means this column is one of the output columns. Some columns may not be output columns but still need to read because it has filters.